src/dbnode/storage/index/mutable_segments.go

```diff
@@ -761,12 +790,11 @@ func (m *mutableSegments) backgroundCompactWithTask(
 	var documentsFilter segment.DocumentsFilter
 	if gcRequired {
 		// Only actively filter out documents if GC is required.
-		documentsFilter = m.seriesActiveFn
+		documentsFilter = segment.DocumentsFilter(m)
```

Comment:
I think we might want to have this filter not be nil even with no `gcRequired`, so that we can still apply the `OnDuplicateDoc` logic